examples/SampleApp/src/hooks/usePaginatedSearchedMessages.ts

```diff
@@ -28,7 +28,7 @@ export const usePaginatedSearchedMessages = (
 ) => {
   const [loading, setLoading] = useState(true);
   const [refreshing, setRefreshing] = useState(false);
-  const [error, setError] = useState(null);
+  const [error, setError] = useState<Error | boolean>(false);
```

Comment:
I know it's not introduced by this PR necessarily, but this handling of errors as either an `Error` or a `boolean` is quite the code smell, we really should do something about it as it's very likely to give us trouble in the future.

After having worked with it, do you have any decent idea of _why_ we have it like this and can't just set an error to be of the Error type? What can we do to make it better? Does something need to change in the javascript client do you think, or is this only local to our SDK?

Reply:
Just the typescript thingy I would say - https://fettblog.eu/typescript-typing-catch-clauses/

Reply:
Also @khushal87 appreciate that you tried to fix it, but this change/fix is unrelated to this PR. So always better to raise a separate PR. Especially since this PR is quite huge and having unrelated changes doesn't help do code review.

Reply:
Just seems really odd for it to be boolean 😳 But I can make a PR to change it and see if it seems alright to do so

Reply:
Honestly, this entire change was not at all related to the PR after I started with the branch this came up as a merge conflict on my PR. So, I fixed using the same change as we decided in this PR - https://github.com/GetStream/stream-chat-react-native/pull/905.
Even I would never introduce it in the branch, but while resolving the merge conflict I had done this. Plus, when I took the pull both the files were giving typescript error in the catch block.